experimental/displaynames/tests/tests.rs

```diff
@@ -0,0 +1,197 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[allow(unused_imports)]
+use icu_displaynames::{DisplayNamesOptions, Fallback, LanguageDisplay, Style, Capitalization, LocaleDisplayNamesFormatter};
+use writeable::assert_writeable_eq;
+use icu_locid::locale;
+
+macro_rules! generate_locale_displaynames_test {
+    ($test_name: ident, $constructor: ident, $options: expr,
+     [$(($locale:literal, $locale_to_test:literal, $expected:literal)),+ $(,)?]) => {
+        #[test]
+        fn $test_name(){
+            let display_name_obj = LocaleDisplayNamesFormatter::$constructor(
+                &icu_testdata::unstable(),
+                &locale!($locale).into(),
+                $options
+            )
+            .expect("Data should load successfully.");
+
+            $(
+                assert_writeable_eq!(
+                    display_name_obj.of(locale!($locale_to_test)).unwrap_or_default(),
+                    $expected
+                );
+            )+
+        }
+    };
+}
+generate_locale_displaynames_test!(
+    test_locale_dn_with_middle_capitalization,
+    try_new_unstable,
+    DisplayNamesOptions {
+        capitalization: Capitalization::MiddleOfSentence
+    },
+    [
+        ("da", "en", "engelsk"),
+        ("da", "en@calendar=buddhist", "engelsk (buddhistisk kalender)"),
+        ("da", "en_GB", "engelsk (Storbritannien)"),
+        ("es", "en", "ingles"), // With acute on the e
+        ("es", "en_GB", "ingles (Reino Unido)"),
+        ("es", "en", "engelsk"),
+    ]
+);
+
+generate_locale_displaynames_test!(
+    test_locale_dn_with_short_style,
+    try_new_unstable,
+    DisplayNamesOptions {
+        capitalization: Capitalization::MiddleOfSentence,
+        style: Style::Short
+    },
+    [
+        ("da", "en_GB", "engelsk (Storbritannien)"),
+        ("es", "en_GB", "ingles (RU)"),
+    ]
+);
+
+generate_locale_displaynames_test!(
+    test_locale_dn_with_dialect,
+    try_new_unstable,
+    DisplayNamesOptions {
+        capitalization: Capitalization::MiddleOfSentence,
+        language_display: LanguageDisplay::Dialect
+    },
+    [
+        ("da", "en_GB", "britisk engelsk"),
```

Comment:
Thought: It would be nice if we can stick with ICU4X test locales. Outside this particular PR, we should discuss testing in non-testdata locales in light of globaldata.